File: joiner-core/src/test/java/cz/encircled/joiner/test/core/data/TestData.java
package cz.encircled.joiner.test.core.data;

import java.util.Collections;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;

import cz.encircled.joiner.test.model.Address;
import cz.encircled.joiner.test.model.Group;
import cz.encircled.joiner.test.model.Key;
import cz.encircled.joiner.test.model.NormalUser;
import cz.encircled.joiner.test.model.Password;
import cz.encircled.joiner.test.model.SuperUser;
import cz.encircled.joiner.test.model.User;
import org.springframework.stereotype.Component;
import org.springframework.test.annotation.Commit;
import org.springframework.transaction.annotation.Transactional;

/**
 * @author Kisel on 26.01.2016.
 */
@Component
public class TestData {

    @PersistenceContext
    private EntityManager entityManager;

    @Transactional
    @Commit
    public void prepareData() {
        if (entityManager.createQuery("select u from User u").setMaxResults(1).getResultList().size() > 0) {
            return;
        }

        Group group = new Group();
        group.setName("group1");
        entityManager.persist(group);

        baseUserCreate(group, 1, true);
        baseUserCreate(group, 2, true);
        baseUserCreate(group, 2, false);
        superUser(group);
        normalUser(group);

        entityManager.flush();
        entityManager.clear();
    }

    private void normalUser(Group group) {
        NormalUser user = new NormalUser();
        user.setGroups(Collections.singletonList(group));
        user.setName("normalUser1");
        entityManager.persist(user);

        Password password = new Password();
        password.setName("normalUser1password1");
        password.setNormalUser(user);
        entityManager.persist(password);

        Address address = new Address();
        address.setName("normalUser1street1");
        address.setUser(user);
        entityManager.persist(address);
    }

    private void superUser(Group group) {
        Key key = new Key();
        key.setName("key1");
        entityManager.persist(key);

        SuperUser superUser = new SuperUser();
        superUser.setName("superUser1");
        superUser.setGroups(Collections.singletonList(group));
        superUser.setKey(key);
        entityManager.persist(superUser);
    }

    private void baseUserCreate(Group group, int index, boolean withAddresses) {
        User user = new User();
        user.setName("user" + index);
        user.setGroups(Collections.singletonList(group));
        entityManager.persist(user);

        if (withAddresses) {
            Address address = new Address();
            address.setName("user" + index + "street1");
            address.setUser(user);

            Address address2 = new Address();
            address2.setName("user" + index + "street2");
            address2.setUser(user);
            entityManager.persist(address);
            entityManager.persist(address2);
        }
    }


}

File: joiner-core/src/test/java/cz/encircled/joiner/test/core/data/TestData.kt
package cz.encircled.joiner.test.core.data

import cz.encircled.joiner.test.model.*
import org.springframework.stereotype.Component
import org.springframework.test.annotation.Commit
import org.springframework.transaction.annotation.Transactional
import javax.persistence.EntityManager
import javax.persistence.PersistenceContext

/**
 * @author Kisel on 26.01.2016.
 */
@Component
class TestData {

    @PersistenceContext
    private val entityManager: EntityManager? = null

    @Transactional
    @Commit
    fun prepareData() {
        if (entityManager!!.createQuery("select u from User u").setMaxResults(1).resultList.size > 0) {
            return
        }

        val group = Group()
        group.name = "group1"
        entityManager.persist(group)

        baseUserCreate(group, 1, true)
        baseUserCreate(group, 2, true)
        baseUserCreate(group, 2, false)
        superUser(group)
        normalUser(group)

        entityManager.flush()
        entityManager.clear()
    }

    private fun normalUser(group: Group) {
        val user = NormalUser()
        user.groups = listOf(group)
        user.name = "normalUser1"
        entityManager!!.persist(user)

        val password = Password()
        password.name = "normalUser1password1"
        password.normalUser = user
        entityManager.persist(password)

        val address = Address()
        address.name = "normalUser1street1"
        address.user = user
        entityManager.persist(address)

        val superUser = SuperUser()
        superUser.name = "superUser2"
        superUser.groups = listOf(group)
        entityManager.persist(superUser)

        val contact = Contact()
        contact.name = "PhoneNumber"
        contact.employmentUser = user
        contact.user = superUser
        entityManager.persist(contact)
    }

    private fun superUser(group: Group) {
        val key = Key()
        key.name = "key1"
        entityManager!!.persist(key)

        val superUser = SuperUser()
        superUser.name = "superUser1"
        superUser.groups = listOf(group)
        superUser.key = key
        entityManager.persist(superUser)
    }

    private fun baseUserCreate(group: Group, index: Int, withAddresses: Boolean) {
        val user = User()
        user.name = "user" + index
        user.groups = listOf(group)
        entityManager!!.persist(user)

        if (withAddresses) {
            val address = Address()
            address.name = "user" + index + "street1"
            address.user = user

            val address2 = Address()
            address2.name = "user" + index + "street2"
            address2.user = user
            entityManager.persist(address)
            entityManager.persist(address2)
        }
    }


}
